contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,150 @@
+        _token.delegate(_delegatee);
+    }
+}
+
+/**
+ * @dev A utility contract to let delegators to pick multiple delegatee
+ */
+contract ERC20MultiDelegate is ERC1155, Ownable {
+    using Address for address;
+
+    ERC20Votes token;
+
+    /**
+     * @dev Constructor.
+     * @param _token The ERC20 token address
+     */
+    constructor(ERC20Votes _token) ERC1155("http://some.metadata.url/{id}") {
+        token = _token;
+    }
+
+    /**
+     * @dev Public method for the delegation of multiple delegatees.
+     * @param delegatees List of delegatee addresses
+     * @param amounts ERC20 voting power amount to be distributed among delegatees
+     */
+    function delegateMulti(
+        address[] calldata delegatees,
+        uint256[] calldata amounts
+    ) external {
+        require(
+            delegatees.length > 0,
+            "You should pick at least one delegatee"
+        );
+        require(
+            delegatees.length == amounts.length,
+            "Amounts should be defined for each delegatee"
+        );
+
+        uint256[] memory ids = new uint256[](delegatees.length);
+
+        for (uint256 index = 0; index < delegatees.length; ) {
+            address delegatee = delegatees[index];
+            // creates a proxy delegator contract with deterministic address
+            // salt occurs from delegatee address + sender address
+            bytes memory bytecode = getBytecode(token, delegatee);
+            bytes32 salt = keccak256(abi.encodePacked(delegatee, msg.sender));
+            address predeterminedProxyAddress = getAddress(
+                bytecode,
+                uint256(salt)
+            );
+            // transfer ERC20 for the proxy delegation
+            // initialize the contract after with predetermined address
+            token.transferFrom(
+                msg.sender,
+                predeterminedProxyAddress,
+                amounts[index]
+            );
+            new ERC20ProxyDelegator{salt: salt}(
+                token,
+                delegatee
+            );
+
+            ids[index] = uint256(uint160(delegatee));
+            unchecked {
+                index++;
+            }
+        }
+        _mintBatch(msg.sender, ids, amounts, "");
+    }
+
+    /**
+     * @dev Public method to withdraw ERC20 voting power from proxy delegators to the actual delegator
+     */
+    function withdraw(address[] calldata delegatees) external {
+        for (uint256 index = 0; index < delegatees.length; ) {
+            // get the delegatee list from user
+            address delegatee = delegatees[index];
+            // PDT - proxy delegation token
+            // check if user has the PDT for the provided delegatees
+            uint256 amount = ERC1155(this).balanceOf(
+                msg.sender,
+                uint256(uint160(delegatee))
+            );
+            // burn PDT's
+            _burn(msg.sender, uint256(uint160(delegatee)), amount);
```

Comment:
Should probably use `burnBatch` here, and put it at the end to avoid reentrancy concerns.